metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py

```diff
@@ -1,6 +1,7 @@
 import dataclasses
 import logging
 import os
+import pathlib
```

Comment:
Why are we putting so much code in __init__.py? Please move this into another file and just have the relevant imports in __init__.py from the file.